python/ray/serve/_private/replica.py

```diff
@@ -692,44 +641,140 @@ async def update_user_config(self, user_config: Any):
                 )
                 await reconfigure_method(user_config)
 
-    async def handle_request(
+    @asynccontextmanager
+    async def wrap_user_method_call(
+        self,
+        request_metadata: RequestMetadata,
+    ):
+        """Context manager that should be used to wrap user method calls.
+
+        This sets up the serve request context, grabs the reader lock to avoid mutating
+        user_config during method calls, and records metrics based on the result of the
+        method.
+        """
+        # Set request context variables for subsequent handle so that
+        # handle can pass the correct request context to subsequent replicas.
+        ray.serve.context._serve_request_context.set(
+            ray.serve.context.RequestContext(
+                request_metadata.route,
+                request_metadata.request_id,
+                self.app_name,
+                request_metadata.multiplexed_model_id,
+            )
+        )
+
+        logger.info(
+            f"Started executing request {request_metadata.request_id}",
+            extra={"log_to_stderr": False},
+        )
+        start_time = time.time()
+        user_exception = None
+        try:
+            # TODO: WTF is up with this thing.
```

Comment:
Does this still applies? lol

Reply:
lol it does at the moment, it's related to the bug I linked above